packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -150,29 +150,37 @@ export const render = async (req, res, next) => {
         })
     }
 
+    // Serialize the application extensions that have an async getRoutes where the key is the name of the extension
+    // and the value is the serialized extension data.
+    const serializedExtensions = Object.fromEntries(
+        applicationExtensions
+            .filter((extension) => extension.getRoutesAsync)
```

Comment:
🤔 Should react sdk be aware to check for `getRoutesAsync`? What if this condition changes in the future? Then the react sdk needs to be updated too.

I think it's better that the responsibility lies solely within the app extension.